lib/fastlane/plugin/wpmreleasetoolkit/helper/github_helper.rb

```diff
@@ -8,38 +8,23 @@ module Fastlane
 
   module Helper
     class GithubHelper
-      def self.github_token!
-        token = [
-          'GHHELPER_ACCESS', # For historical reasons / backward compatibility
-          'GITHUB_TOKEN',    # Used by the `gh` CLI tool
-        ].map { |key| ENV[key] }
-                .compact
-                .first
-
-        token || UI.user_error!('Please provide a GitHub authentication token via the `GITHUB_TOKEN` environment variable')
-      end
-
-      # Creates a client for the GitHub API
+      # Helper for GitHub Actions
       #
-      # @param [String] accesstoken GitHub OAuth access token
+      # @param [String?] github_token GitHub OAuth access token
       #
-      def self.github_client(accesstoken)
-        @@client ||= begin
-          client = Octokit::Client.new(access_token: accesstoken)
-
-          # Fetch the current user
-          user = client.user
-          UI.message("Logged in as: #{user.name}")
+      def initialize(github_token:)
+        @client = Octokit::Client.new(access_token: github_token)
 
-          # Auto-paginate to ensure we're not missing data
-          client.auto_paginate = true
+        # Fetch the current user
+        user = @client.user
+        UI.message("Logged in as: #{user.name}")
 
-          client
-        end
+        # Auto-paginate to ensure we're not missing data
+        @client.auto_paginate = true
       end
 
       def self.get_milestone(repository, release)
-        miles = github_client(github_token!).list_milestones(repository)
+        miles = @client.list_milestones(repository)
```

Comment:
Note that if you go with the route of declaring `attr_reader :client` (as an instance property accessible from outside), you would be able to either use `@client` or `client` in those internal implementations of the helper methods here — both would work —, but I think the Ruby conventions when an `attr_reader` is declared is then to try to prefer using the attribute / accessor (`client`) over the instance variable (`@client`) 🤷 

(If you've worked on an ObjC codebase before, an `attr_reader` declaration is similar to `@property(readonly)` and  accessing `@client` vs `client` is similar to accessing `self.client` vs `_client` 😄 )